Source/Core/DolphinQt/GameList/GameList.cpp

```diff
@@ -628,6 +637,48 @@ void GameList::OpenGCSaveFolder()
     ModalMessageBox::information(this, tr("Information"), tr("No save data found."));
 }
 
+#ifdef _WIN32
+void GameList::AddShortcutToDesktop()
+{
+  const auto game = GetSelectedGame();
+  const auto game_name = game->GetLongName();
+  const auto file_path = game->GetFilePath();
+  const auto file_name = game->GetFileName();
+
+  std::wstring args = UTF8ToTStr("-e \"" + file_path + "\"");
+  std::wstring dolphin_path = QCoreApplication::applicationFilePath().toStdWString();
+
+  // Gets the desktop position with shorcut
+  WCHAR* desktop = nullptr;
+  SHGetKnownFolderPath(FOLDERID_Desktop, KF_FLAG_NO_ALIAS, NULL, &desktop);
+  std::wstring desktop_path = std::wstring(desktop) + UTF8ToTStr("\\" + game_name + ".lnk");
+
+  HRESULT hres;
+  Microsoft::WRL::ComPtr<IShellLink> psl;
+
+  CoInitialize(NULL);
+  hres = CoCreateInstance(CLSID_ShellLink, nullptr, CLSCTX_INPROC_SERVER, IID_PPV_ARGS(&psl));
+
+  if (SUCCEEDED(hres))
```

Comment:
Consider inverting the conditional to return early on failure to avoid unneeded indentation.